lib/src/options/union_options.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:data_class_plugin/data_class_plugin.dart';
+import 'package:data_class_plugin/src/options/extensions.dart';
+import 'package:data_class_plugin/src/options/options_config.dart';
+
+@DataClass()
+class UnionOptions {
+  /// Shorthand constructor
+  const UnionOptions({
+    this.optionsConfig = const <String, OptionConfig>{},
+  });
+
+  final Map<String, OptionConfig> optionsConfig;
+
+  bool effectiveDataClass(String filePath) =>
+      optionsConfig.effectiveDataClass(filePath: filePath, defaultValue: true);
+
+  bool effectiveFromJson(String filePath) =>
```

Comment:
I think we need to make the fromJson/toJson default value as false, to follow the same logic of data classes